### PR TITLE
Updates to local setup scripts and introduce new local setup scripts for non-gardener use cases

### PR DESCRIPTION
**Pull request by unmarshall**
**What this PR does / why we need it**:
Creates separate local setup scripts for gardener and non-gardener use cases.
There are now 4 scripts that are made available:
For gardener specific setup/restore following scripts have been introduced:
```bash
./hack/gardener_local_setup.sh -h                                                                                                        git:script

    Usage: gardener_local_setup.sh [Options]
    Options:
      -d | --seed                       <seed-cluster-name>                 (Required) Name of the Gardener Seed Cluster
      -t | --shoot                      <shoot-cluster-name>                (Required) Name of the Gardener Shoot Cluster
      -p | --project                    <project-name>                      (Required) Name of the Gardener Project
      -i | --provider                   <provider-name>                     (Required) Infrastructure provider name. Supported providers (gcp|aws|azure|vsphere|openstack|alicloud|metal|equinix-metal)
      -l | --landscape                  <landscape-name>                    (Optional) Name of the landscape. Defaults to dev
      -k | --kubeconfig-path            <relative-kubeconfig-path>          (Optional) Relative path to the <PROJECT-DIR> where kubeconfigs will be downloaded. Path should not start with '/'. Default: <PROJECT-DIR>/dev/kube-configs
      -m | --mcm-provider-project-path  <absolute-mcm-provider-project-dir> (Optional) MCM Provider project directory. If not provided then it assumes that both mcm and mcm-provider projects are under the same parent directory
      -e | --kubeconfig-expiry-seconds  <expiry-duration-in-seconds>        (Optional) Common expiry durations in seconds for control and target KubeConfigs. Default: 3600 seconds
```
```bash
./hack/gardener_local_restore.sh -h                                                                                                      git:script

    Usage: gardener_local_restore.sh [Options]
    Options:
      -t | --shoot                      <shoot-cluster-name>                (Required) Name of the Gardener Shoot Cluster
      -p | --project                    <project-name>                      (Required) Name of the Gardener Project
      -l | --landscape                  <landscape-name>                    (Optional) Name of the landscape. Defaults to dev
      -i | --provider                   <provider-name>                     (Required) Infrastructure provider name. Supported providers (gcp|aws|azure|vsphere|openstack|alicloud|metal|equinix-metal)
      -k | --kubeconfig-path            <relative-kubeconfig-path>          (Optional) Relative path to the <PROJECT-DIR> where kubeconfigs will be downloaded. Path should not start with '/'. Default: <PROJECT-DIR>/dev/kube-configs
      -m | --mcm-provider-project-path  <absolute-mcm-provider-project-dir> (Optional) MCM Provider project directory. If not provided then it assumes that both mcm and mcm-provider projects are under the same parent directory
```

`Makefile` targets have the following changes:
```bash
> make gardener-setup
> make gardener-restore
> make gardener-local-mcm-up
```

For non-gardener use cases the following scripts have been introduced:
```bash
./hack/non_gardener_local_setup.sh -h                                                                                                    git:script

    Usage: non_gardener_local_setup.sh [Options]
    Options:
      -n | --namespace                  <namespace>                         (Required) This is the namespace where MCM pods are deployed.
      -c | --control-kubeconfig-path    <control-kubeconfig-path>           (Required) Kubeconfig file path which points to the control-plane of cluster where MCM is running.
      -t | --target-kubeconfig-path     <target-kubeconfig-path>            (Required) Kubeconfig file path which points to control plane of the cluster where nodes are created.
      -i | --provider                   <provider-name>                     (Required) Infrastructure provider name. Supported providers (gcp|aws|azure|vsphere|openstack|alicloud|metal|equinix-metal)
      -m | --mcm-provider-project-path  <absolute-mcm-provider-project-dir> (Optional) MCM Provider project directory. If not provided then it assumes that both mcm and mcm-provider projects are under the same parent directory
```
```bash
./hack/non_gardener_local_restore.sh -h                                                                                                  git:script

    Usage: non_gardener_local_restore.sh [Options]
    Options:
      -n | --namespace                  <namespace>                         (Required) This is the namespace where MCM pods are deployed.
      -c | --control-kubeconfig-path    <control-kubeconfig-path>           (Required) Kubeconfig file path which points to the control-plane of cluster where MCM is running.
      -i | --provider                   <provider-name>                     (Required) Infrastructure provider name. Supported providers (gcp|aws|azure|vsphere|openstack|alicloud|metal|equinix-metal)
      -m | --mcm-provider-project-path  <absolute-mcm-provider-project-dir> (Optional) MCM Provider project directory. If not provided then it assumes that both mcm and mcm-provider projects are under the same parent directory
```
`Makefile` targets have the following changes:
```bash
> make non-gardener-setup
> make non-gardener-restore
> make non-gardener-local-mcm-up
```

**Which issue(s) this PR fixes**:
Fixes Part of #846

**Special notes for your reviewer**:

**Release note**:

```other operator
Makefile targets have changed: Introduced gardener-setup, gardener-restore, gardener-local-mcm-up, non-gardener-setup, non-gardener-restore,  non-gardener-local-mcm-up. Users can also directly use the scripts which are used by these makefile targets.
```

### CODE REVIEW

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Review by elankath — COMMENTED** [review]
Please update `development/local_setup.md` if possible. Need to roll this out to new joinees. If you don't have time, I can do this.

File: Makefile [hunk 1]
```diff
@@ -42,21 +43,53 @@ setup:
...
+gardener-local-mcm-up: gardener-setup
 	$(MAKE) start;
 
-.PHONY: restore
-restore:
+.PHONY: gardener-restore
+gardener-restore:
 	@echo "enter project name"; \
 	read PROJECT; \
 	echo "enter shoot name"; \
```
Comment (himanshu-kun) [inline Makefile:56]: it doesn't ask for `SEED`

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
could you also remove `hack/admin-kube-config-request.json`

File: Makefile (hunk 1 above)
Comment (unmarshall) [reply Makefile:56]: the script does not need it. Where do you think seed should be used?

**Comment by unmarshall** [comment]
>could you also remove hack/admin-kube-config-request.json

This file gets removed when you can  the gardener_restore.sh script
```
  echo "Removing generated admin kube config json..."
  rm -f "${SCRIPT_DIR}"/admin-kube-config-request.json
```
But i can also remove it once the kube configs have been downloaded successfully.

**Comment by unmarshall** [comment]
>Is the development/local_setup.md already updated showing how to use these scripts ?

I have updated the local_setup.md now. PTAL.

File: Makefile (hunk 1 above)
Comment (himanshu-kun) [reply Makefile:56]: ok you are targetting the control plane directly 
```
gardenctl target --garden "${virtual_garden_cluster}" --project "${PROJECT}" --shoot "${SHOOT}" --control-plane
```

**Review by himanshu-kun — APPROVED** [review]
working well for me now
Thanks for the changes @unmarshall !

**Comment by himanshu-kun** [comment]
/needs second-opinion 
waiting till @elankath re-reviews

**Review by elankath — APPROVED** [review]
Thanks for writing doc. We need to clean the IT setup to avoid the symbolic link. let the path to MCM repo be given in the new `.env` file. This can be taken up later, not now.

/lgtm

**Pull request merged**